packages/udocker/0005-set-variable-for-containers.patch

```diff
@@ -0,0 +1,11 @@
+diff -ur orig/udocker/engine/base.py mod/udocker/engine/base.py
+--- orig/udocker/engine/base.py	2024-08-28 16:01:10.000000000 +0200
++++ mod/udocker/engine/base.py	2025-05-12 21:06:27.573586252 +0200
+@@ -559,6 +559,7 @@
+         else:
+             self.opt["env"].append(r"PS1=%s\$ " % self.container_id[:8])
+ 
++        self.opt["env"].append("XDG_RUNTIME_DIR=/tmp/runtime-root")
```

Comment:
Are you sure this is correct?
`runtime-root` seems like a placeholder here.

Reply:
I'm not. I can generally set /tmp actually. It's simply a passed variable to the container. It is not required, however many applications require XDG_RUNTIME_DIR.

Reply:
/tmp cannot be correct in any case, there is no /tmp on android. @TERMUX_PREFIX@/tmp/runtime-root would work though..

Reply:
Or is this path only for when inside proot?

Reply:
Well that depends how important it is for `XDG_RUNTIME_DIR` to be set up inside the container.
A lot of distros will set up XDG basedirs as part bringing up userspace, so it should either already be set by the distro, or there should be a fallback in place on the application's end.

It would be nice to provide a sensible default, but it doesn't seem like upstream does, so I don't think it's critically important.

Reply:
There is no default given in the spec for `XDG_RUNTIME_DIR`[^1], however `/run/user/${UID}` is common on "regular" Linux distros.
That directory may or may not exist in this context.
And in Termux we usually treat `${PREFIX}/var/run/user` as the `XDG_RUNTIME_DIR`.

[^1]: https://specifications.freedesktop.org/basedir-spec/latest/#variables:~:text=should%20be%20used.-,%24XDG_RUNTIME_DIR,-defines%20the%20base

Reply:
Removed